content/configuration/endpoints.md

```diff
@@ -0,0 +1,205 @@
+
+The standard way to expose your services to the internet is to use the following configuration:
+ 
+``` yaml
+tasks:
+    deployment:
+        deploy:
+            # ...
+            services:
+                web:
+                    specification:
+                        accessibility:
+                            from_external: false
+```
+
+This is fine for most purposes, but has it has three main limitations: 
+
+**1)** It presents difficulties when trying to whitelist IP addresses for authentication as the client IP address is not passed through to the service. This can be resolved by using an annotation, which ensures that the client IP address is passed through to the service: 
+
+``` yaml
+tasks:
+    deployment:
+        deploy:
+            # ...
+            services:
+                web:
+                    endpoints:
+                      - name: www
+                        annotations:
+                          service.beta.kubernetes.io/external-traffic: OnlyLocal
+                    specification:
+                        accessibility:
+                            from_external: false
+```
+
+**2)** The IP addresses are chosen from a random pool and by default are not static, which means they cannot be reliably associated with a DNS record.
+
+**3)** It uses a public IP address per service, which can be expensive. 
+
+## Using a Load Balancer
+
+Most cloud services offer the option of using a load balancer, which mitigates two of the above limitations - with a load balancer public IP addresses are passed through to the service, and the IP address of the load balancer is static so can be set up in a DNS record. However, using load balancers can also be expensive as you still use a public IP address for each service.
+
+## Using the Nginx Ingress Controller
+
+Using the [Nginx Ingress controller](https://github.com/kubernetes/ingress/tree/master/controllers/nginx) presents an option that mitigates all three of the above limitations, as it only requires one public IP address - a series of rules can be used to determine how inbound connections are directed to your services. The Nginx Ingress controller also allows SSL to be configured in one place so removes the need to configure SSL for each service independently.
+
+### Installation
+
+ContinuousPipe currently provides no way to create the Nginx Ingress controller - this needs to be done as a manual step when setting up your Kubernetes cluster. Once the Ingress is deployed, it will be assigned an IP address from a random pool, which will then be static for the lifetime of the Ingress. The IP address of the manually deployed Ingress then needs to be set up in a DNS rule so that it is associated with your domain, unless you are using [Cloudflare integration](#nginx-ingress-with-cloudflare-ssl).
+
+**Instructions for deploying the Ingress and retrieving the IP address for use in DNS are available here:** https://github.com/continuouspipe/nginx-ingress-controller.
```

Comment:
Actually, I would point to Kubernetes documentation. It's about deploying an "ingress controller": https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-controllers